sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -26,9 +26,12 @@ trait FunctionRegistry {
 
   def registerFunction(name: String, builder: FunctionBuilder): Unit
 
-  def lookupFunction(name: String, children: Seq[Expression]): Expression
-
   def caseSensitive: Boolean
+
+  def lookupFunction(
+      name: String,
+      children: Seq[Expression],
+      distinct: Boolean = false): Expression
```

Comment:
In Spark SQL, we put everything about the expression, as its constructor argument list, but Hive requires some additional object to keep the expression info, and both of them will be initialized/used within the executors. We need to throw exception if it's not an UDAF once `distinct` was set to true.